packages/oc-webpack/test/oc-webpack.test.js

```diff
@@ -42,22 +44,96 @@ test('webpack compiler', done => {
     )
   });
 
-  webpackCompiler(config, (warning, serverContentBundled) => {
+  webpackCompiler(config, (error, serverContentBundled) => {
     expect(serverContentBundled).toMatchSnapshot();
     done();
   });
 });
 
-test('webpack compiler with error', done => {
+test('webpack compiler verbose', done => {
+  const loggerMock = { log: jest.fn() };
+  const config = webpackConfigurator({
+    logger: loggerMock,
+    stats: 'verbose',
+    dependencies: { lodash: '' },
+    publishFileName: 'server.js',
+    serverPath: path.join(
+      __dirname,
+      '../../../mocks/jade-component',
+      'server.js'
+    )
+  });
+
+  webpackCompiler(config, (error, serverContentBundled) => {
+    const consoleOutput = loggerMock.log.mock.calls[0][0];
+    expect(serverContentBundled).toMatchSnapshot();
+    expect(consoleOutput).toMatch(/Hash:(.*?)01e93c95dfecf93de280/);
+    expect(consoleOutput).toMatch(/Entrypoint(.*?)main(.*?)=(.*?)server.js/);
+    expect(consoleOutput).toMatch(/external \"lodash\"/);
+    done();
+  });
+});
+
+test('webpack compiler with fatal error', done => {
+  const config = webpackConfigurator({
+    stats: 'errors-only',
+    dependencies: { lodash: '' },
+    publishFileName: 'server.js',
+    serverPath: path.join(
+      __dirname,
+      '../../../mocks/jade-component',
+      'server.js'
+    )
+  });
+
+  config.plugins.push(function() {
+    this.plugin('run', (compiler, cb) =>
+      cb('This is a fatal compilation error')
+    );
+  });
+
+  webpackCompiler(config, (error, serverContentBundled) => {
+    expect(error).toMatchSnapshot();
+    done();
+  });
+});
+
+test('webpack compiler with soft error', done => {
   const config = webpackConfigurator({
     stats: 'errors-only',
     dependencies: { lodash: '' },
     publishFileName: 'server.js',
     serverPath: path.join(__dirname, 'some/not/valid/path', 'server.js')
   });
 
-  webpackCompiler(config, (warning, serverContentBundled) => {
-    expect(warning).toContain(`Entry module not found: Error: Can't resolve`);
+  webpackCompiler(config, (error, serverContentBundled) => {
+    expect(error).toContain(`Entry module not found: Error: Can't resolve`);
+    done();
+  });
+});
+
+test('webpack compiler with warning', done => {
+  const loggerMock = { log: jest.fn() };
+  const config = webpackConfigurator({
+    logger: loggerMock,
+    stats: 'errors-only',
+    dependencies: { lodash: '' },
+    publishFileName: 'server.js',
+    serverPath: path.join(
+      __dirname,
+      '../../../mocks/jade-component',
+      'server.js'
+    )
+  });
+
+  config.plugins.push(function() {
+    this.plugin('run', (compiler, cb) => cb.call(compiler));
+    this.plugin('done', stats => stats.compilation.warnings.push('A warning'));
+  });
+
+  webpackCompiler(config, (error, serverContentBundled) => {
+    expect(loggerMock.log.mock.calls[0][0]).toContain('A warning');
```

Comment:
as discussed with @nickbalestra - this is a breaking change - but we can consider it as patch fix I guess